src/main/java/de/azapps/kafkabackup/restore/message/RestoreMessageService.java

```diff
@@ -172,29 +172,30 @@ private String partitionMessageWriterWorkersInfo() {
     final int partitionNumber;
     private MessageRestorationStatus messageRestorationStatus;
     private Map<Long, RestoredMessageInfo> restoredMessageInfoMap;
-    private long maxOriginalOffset = 0L;
+    private long maxOriginalOffset = -1L;
 
     public TopicPartitionToRestore(TopicConfiguration topicConfiguration, int partitionNumber) {
       this.topicConfiguration = topicConfiguration;
       this.partitionNumber = partitionNumber;
       this.messageRestorationStatus = MessageRestorationStatus.WAITING;
       this.restoredMessageInfoMap = new HashMap<>();
+      // Start out by always mapping offset 0 to offset 0 (for empty topics)
+      addRestoredMessageInfo(0L, 0L);
     }
 
     public String getTopicPartitionId() {
       return topicConfiguration.getTopicName() + "." + partitionNumber;
     }
 
-    public void addRestoredMessageInfo(long originalOffset, byte[] key, Long newOffset) {
-      restoredMessageInfoMap.put(originalOffset, new RestoredMessageInfo(originalOffset, key, newOffset));
+    public void addRestoredMessageInfo(long originalOffset, Long newOffset) {
+      restoredMessageInfoMap.put(originalOffset, new RestoredMessageInfo(originalOffset, newOffset));
       maxOriginalOffset = Math.max(maxOriginalOffset, originalOffset);
     }
   }
 
   @Data
   public static class RestoredMessageInfo {
     private final long originalOffset;
-    private final byte[] key;
```

Comment:
This is unused, and couldn't be initialised for the default mapping. So simplified things a bit.

Reply:
Since we would have map from oldOffset to object containing oldOffset and newOffset I've changed it to simply map old offset to new offset and removed RestoredMessageInfo completely.